// @aws-sdk/client-textract/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultTextractHttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { AnalyzeDocument$, AnalyzeExpense$, AnalyzeID$, CreateAdapter$, CreateAdapterVersion$, DeleteAdapter$, DeleteAdapterVersion$, DetectDocumentText$, GetAdapter$, GetAdapterVersion$, GetDocumentAnalysis$, GetDocumentTextDetection$, GetExpenseAnalysis$, GetLendingAnalysis$, GetLendingAnalysisSummary$, ListAdapters$, ListAdapterVersions$, ListTagsForResource$, StartDocumentAnalysis$, StartDocumentTextDetection$, StartExpenseAnalysis$, StartLendingAnalysis$, TagResource$, UntagResource$, UpdateAdapter$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
__exportStar(require("./models/errors"), exports);
const { TextractServiceException } = require("./models/TextractServiceException");
exports.TextractServiceException = TextractServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "textract",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class TextractClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultTextractHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AnalyzeDocumentCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "AnalyzeDocument", {})
    .n("TextractClient", "AnalyzeDocumentCommand")
    .sc(AnalyzeDocument$)
    .build() {
}

class AnalyzeExpenseCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "AnalyzeExpense", {})
    .n("TextractClient", "AnalyzeExpenseCommand")
    .sc(AnalyzeExpense$)
    .build() {
}

class AnalyzeIDCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "AnalyzeID", {})
    .n("TextractClient", "AnalyzeIDCommand")
    .sc(AnalyzeID$)
    .build() {
}

class CreateAdapterCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "CreateAdapter", {})
    .n("TextractClient", "CreateAdapterCommand")
    .sc(CreateAdapter$)
    .build() {
}

class CreateAdapterVersionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "CreateAdapterVersion", {})
    .n("TextractClient", "CreateAdapterVersionCommand")
    .sc(CreateAdapterVersion$)
    .build() {
}

class DeleteAdapterCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "DeleteAdapter", {})
    .n("TextractClient", "DeleteAdapterCommand")
    .sc(DeleteAdapter$)
    .build() {
}

class DeleteAdapterVersionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "DeleteAdapterVersion", {})
    .n("TextractClient", "DeleteAdapterVersionCommand")
    .sc(DeleteAdapterVersion$)
    .build() {
}

class DetectDocumentTextCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "DetectDocumentText", {})
    .n("TextractClient", "DetectDocumentTextCommand")
    .sc(DetectDocumentText$)
    .build() {
}

class GetAdapterCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetAdapter", {})
    .n("TextractClient", "GetAdapterCommand")
    .sc(GetAdapter$)
    .build() {
}

class GetAdapterVersionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetAdapterVersion", {})
    .n("TextractClient", "GetAdapterVersionCommand")
    .sc(GetAdapterVersion$)
    .build() {
}

class GetDocumentAnalysisCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetDocumentAnalysis", {})
    .n("TextractClient", "GetDocumentAnalysisCommand")
    .sc(GetDocumentAnalysis$)
    .build() {
}

class GetDocumentTextDetectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetDocumentTextDetection", {})
    .n("TextractClient", "GetDocumentTextDetectionCommand")
    .sc(GetDocumentTextDetection$)
    .build() {
}

class GetExpenseAnalysisCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetExpenseAnalysis", {})
    .n("TextractClient", "GetExpenseAnalysisCommand")
    .sc(GetExpenseAnalysis$)
    .build() {
}

class GetLendingAnalysisCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetLendingAnalysis", {})
    .n("TextractClient", "GetLendingAnalysisCommand")
    .sc(GetLendingAnalysis$)
    .build() {
}

class GetLendingAnalysisSummaryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "GetLendingAnalysisSummary", {})
    .n("TextractClient", "GetLendingAnalysisSummaryCommand")
    .sc(GetLendingAnalysisSummary$)
    .build() {
}

class ListAdaptersCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "ListAdapters", {})
    .n("TextractClient", "ListAdaptersCommand")
    .sc(ListAdapters$)
    .build() {
}

class ListAdapterVersionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "ListAdapterVersions", {})
    .n("TextractClient", "ListAdapterVersionsCommand")
    .sc(ListAdapterVersions$)
    .build() {
}

class ListTagsForResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "ListTagsForResource", {})
    .n("TextractClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class StartDocumentAnalysisCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "StartDocumentAnalysis", {})
    .n("TextractClient", "StartDocumentAnalysisCommand")
    .sc(StartDocumentAnalysis$)
    .build() {
}

class StartDocumentTextDetectionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "StartDocumentTextDetection", {})
    .n("TextractClient", "StartDocumentTextDetectionCommand")
    .sc(StartDocumentTextDetection$)
    .build() {
}

class StartExpenseAnalysisCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "StartExpenseAnalysis", {})
    .n("TextractClient", "StartExpenseAnalysisCommand")
    .sc(StartExpenseAnalysis$)
    .build() {
}

class StartLendingAnalysisCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "StartLendingAnalysis", {})
    .n("TextractClient", "StartLendingAnalysisCommand")
    .sc(StartLendingAnalysis$)
    .build() {
}

class TagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "TagResource", {})
    .n("TextractClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class UntagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "UntagResource", {})
    .n("TextractClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateAdapterCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Textract", "UpdateAdapter", {})
    .n("TextractClient", "UpdateAdapterCommand")
    .sc(UpdateAdapter$)
    .build() {
}

const paginateListAdapters = createPaginator(TextractClient, ListAdaptersCommand, "NextToken", "NextToken", "MaxResults");

const paginateListAdapterVersions = createPaginator(TextractClient, ListAdapterVersionsCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AnalyzeDocumentCommand,
    AnalyzeExpenseCommand,
    AnalyzeIDCommand,
    CreateAdapterCommand,
    CreateAdapterVersionCommand,
    DeleteAdapterCommand,
    DeleteAdapterVersionCommand,
    DetectDocumentTextCommand,
    GetAdapterCommand,
    GetAdapterVersionCommand,
    GetDocumentAnalysisCommand,
    GetDocumentTextDetectionCommand,
    GetExpenseAnalysisCommand,
    GetLendingAnalysisCommand,
    GetLendingAnalysisSummaryCommand,
    ListAdaptersCommand,
    ListAdapterVersionsCommand,
    ListTagsForResourceCommand,
    StartDocumentAnalysisCommand,
    StartDocumentTextDetectionCommand,
    StartExpenseAnalysisCommand,
    StartLendingAnalysisCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateAdapterCommand,
};
const paginators = {
    paginateListAdapters,
    paginateListAdapterVersions,
};
class Textract extends TextractClient {
}
createAggregatedClient(commands, Textract, { paginators });

const FeatureType = {
    FORMS: "FORMS",
    LAYOUT: "LAYOUT",
    QUERIES: "QUERIES",
    SIGNATURES: "SIGNATURES",
    TABLES: "TABLES",
};
const AdapterVersionStatus = {
    ACTIVE: "ACTIVE",
    AT_RISK: "AT_RISK",
    CREATION_ERROR: "CREATION_ERROR",
    CREATION_IN_PROGRESS: "CREATION_IN_PROGRESS",
    DEPRECATED: "DEPRECATED",
};
const ContentClassifier = {
    FREE_OF_ADULT_CONTENT: "FreeOfAdultContent",
    FREE_OF_PERSONALLY_IDENTIFIABLE_INFORMATION: "FreeOfPersonallyIdentifiableInformation",
};
const BlockType = {
    CELL: "CELL",
    KEY_VALUE_SET: "KEY_VALUE_SET",
    LAYOUT_FIGURE: "LAYOUT_FIGURE",
    LAYOUT_FOOTER: "LAYOUT_FOOTER",
    LAYOUT_HEADER: "LAYOUT_HEADER",
    LAYOUT_KEY_VALUE: "LAYOUT_KEY_VALUE",
    LAYOUT_LIST: "LAYOUT_LIST",
    LAYOUT_PAGE_NUMBER: "LAYOUT_PAGE_NUMBER",
    LAYOUT_SECTION_HEADER: "LAYOUT_SECTION_HEADER",
    LAYOUT_TABLE: "LAYOUT_TABLE",
    LAYOUT_TEXT: "LAYOUT_TEXT",
    LAYOUT_TITLE: "LAYOUT_TITLE",
    LINE: "LINE",
    MERGED_CELL: "MERGED_CELL",
    PAGE: "PAGE",
    QUERY: "QUERY",
    QUERY_RESULT: "QUERY_RESULT",
    SELECTION_ELEMENT: "SELECTION_ELEMENT",
    SIGNATURE: "SIGNATURE",
    TABLE: "TABLE",
    TABLE_FOOTER: "TABLE_FOOTER",
    TABLE_TITLE: "TABLE_TITLE",
    TITLE: "TITLE",
    WORD: "WORD",
};
const EntityType = {
    COLUMN_HEADER: "COLUMN_HEADER",
    KEY: "KEY",
    SEMI_STRUCTURED_TABLE: "SEMI_STRUCTURED_TABLE",
    STRUCTURED_TABLE: "STRUCTURED_TABLE",
    TABLE_FOOTER: "TABLE_FOOTER",
    TABLE_SECTION_TITLE: "TABLE_SECTION_TITLE",
    TABLE_SUMMARY: "TABLE_SUMMARY",
    TABLE_TITLE: "TABLE_TITLE",
    VALUE: "VALUE",
};
const RelationshipType = {
    ANSWER: "ANSWER",
    CHILD: "CHILD",
    COMPLEX_FEATURES: "COMPLEX_FEATURES",
    MERGED_CELL: "MERGED_CELL",
    TABLE: "TABLE",
    TABLE_FOOTER: "TABLE_FOOTER",
    TABLE_TITLE: "TABLE_TITLE",
    TITLE: "TITLE",
    VALUE: "VALUE",
};
const SelectionStatus = {
    NOT_SELECTED: "NOT_SELECTED",
    SELECTED: "SELECTED",
};
const TextType = {
    HANDWRITING: "HANDWRITING",
    PRINTED: "PRINTED",
};
const ValueType = {
    DATE: "DATE",
};
const AutoUpdate = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const JobStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PARTIAL_SUCCESS: "PARTIAL_SUCCESS",
    SUCCEEDED: "SUCCEEDED",
};

exports.AdapterVersionStatus = AdapterVersionStatus;
exports.AnalyzeDocumentCommand = AnalyzeDocumentCommand;
exports.AnalyzeExpenseCommand = AnalyzeExpenseCommand;
exports.AnalyzeIDCommand = AnalyzeIDCommand;
exports.AutoUpdate = AutoUpdate;
exports.BlockType = BlockType;
exports.ContentClassifier = ContentClassifier;
exports.CreateAdapterCommand = CreateAdapterCommand;
exports.CreateAdapterVersionCommand = CreateAdapterVersionCommand;
exports.DeleteAdapterCommand = DeleteAdapterCommand;
exports.DeleteAdapterVersionCommand = DeleteAdapterVersionCommand;
exports.DetectDocumentTextCommand = DetectDocumentTextCommand;
exports.EntityType = EntityType;
exports.FeatureType = FeatureType;
exports.GetAdapterCommand = GetAdapterCommand;
exports.GetAdapterVersionCommand = GetAdapterVersionCommand;
exports.GetDocumentAnalysisCommand = GetDocumentAnalysisCommand;
exports.GetDocumentTextDetectionCommand = GetDocumentTextDetectionCommand;
exports.GetExpenseAnalysisCommand = GetExpenseAnalysisCommand;
exports.GetLendingAnalysisCommand = GetLendingAnalysisCommand;
exports.GetLendingAnalysisSummaryCommand = GetLendingAnalysisSummaryCommand;
exports.JobStatus = JobStatus;
exports.ListAdapterVersionsCommand = ListAdapterVersionsCommand;
exports.ListAdaptersCommand = ListAdaptersCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.RelationshipType = RelationshipType;
exports.SelectionStatus = SelectionStatus;
exports.StartDocumentAnalysisCommand = StartDocumentAnalysisCommand;
exports.StartDocumentTextDetectionCommand = StartDocumentTextDetectionCommand;
exports.StartExpenseAnalysisCommand = StartExpenseAnalysisCommand;
exports.StartLendingAnalysisCommand = StartLendingAnalysisCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.TextType = TextType;
exports.Textract = Textract;
exports.TextractClient = TextractClient;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateAdapterCommand = UpdateAdapterCommand;
exports.ValueType = ValueType;
exports.paginateListAdapterVersions = paginateListAdapterVersions;
exports.paginateListAdapters = paginateListAdapters;
